src/pytesmo/validation_framework/data_averaging.py

```diff
@@ -0,0 +1,423 @@
+    ) -> pd.DataFrame:
+        """
+        Uses time stability concepts to filter point-measurements (pms). Determines wether the upscaled measurement
+        based on a simple average of all the pms is in sufficient agreement with each pm, and if not eliminates pm
+        from the pool.
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            temporally matched DataFrame
+        r_min: float
+            lower threshold for correlation (Pearson) between upscaled and pm
+        see_max: float
+            upper threshold for standard error of estimate
+        min_n: int
+            minimum number of pms to perform the filtering
+
+        Returns
+        -------
+        filtered: pd.DataFrame
+            filtered input
+        """
+        if len(df.columns) < min_n:
+            return df
+
+        # get a trivial (average) upscale estimate
+        estimate = df.mean(axis=1)
+
+        filter_out = []
+        for n, pm in enumerate(df):
+            pm_values = df[pm]
+            regr = linregress(
+                estimate,
+                pm_values
+            )
+            if regr.rvalue < r_min or regr.intercept_stderr > see_max:
+                filter_out.append(pm)
+
+        filtered = df.drop(filter_out)
+        # todo: This could be done iteratively to calibrate the thresholds
+        if len(filtered.columns) < 2:
+            warnings.warn(
+                "The filtering options are too strict. Returning entire dataframe ..."
+            )
+            return df
+
+        return filtered
+
+    def upscale(self, df, method='average', **kwargs) -> pd.DataFrame:
+        """
+        Return the upscaled Dataframe with the specified method
+
+        **New upscaling methods can be specified here in the lut**
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Dataframe of values to average using method
+        method: str
+            averaging method
+        kwargs: keyword arguments
+            Arguments for some upscaling functions
+
+        Returns
+        -------
+        upscaled: pandas.DataFrame
+            dataframe with "upscaled" column
+        """
+        up_function = {
+            "average": self.faverage,
+        }
+        f = up_function[method]
+        # check that column names are all the same and return first
+        g = groupby(df.columns)
+        assert (next(g, True) and not next(g, False))
+        var_name = df.columns[0]
+
+        upscaled = pd.DataFrame(
+            data=f(df, **kwargs),
+            columns=[var_name]
+        )
+
+        return upscaled
+
+    @staticmethod
+    def faverage(df) -> pd.Series:
+        """Simple average of each column in the dataframe"""
+        # workaround to avoid changes in place
+        out = df.mean(axis=1)
+
+        return out
+
+    def wrapper(
+            self,
+            gpi,
+            other_name,
+            tmatching="rescale",
+            up_method="average",
```

Comment:
Can you call this "upscaling_method"? It's a bit more verbose. Probably in the line above "temporal_matching_method" would also be better.